ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -74,4 +77,35 @@ public void shouldFailWhenSyntaxIsInvalid() throws Exception {
     ksqlEngine.createQueries("blah;");
   }
 
+  @Test
+  public void shouldUpdateReferentialIntegrityTableCorrectly() throws Exception {
+    final List<QueryMetadata> queries
+        = ksqlEngine.createQueries("create table bar as select * from test2;" +
+                                   "create table foo as select * from test2;");
+    MetaStore metaStore = ksqlEngine.getMetaStore();
+    Assert.assertTrue(metaStore.getSourceForQuery("TEST2").contains("CTAS_BAR"));
```

Comment:
`assertThat(metaStore.getSourceForQuery(..), equalTo(Utils.mkSet(...))`

similarly for sink